javaslang/src/main/java/javaslang/collection/JavaConverters.java

```diff
@@ -0,0 +1,577 @@
+            return delegate.hashCode();
+        }
+
+        @Override
+        public String toString() {
+            return delegate.toJavaList().toString();
+        }
+
+        // -- private helpers
+
+        private T setDelegateAndGetPreviousElement(int index, Seq<T> newDelegate) {
+            final T previousElement = delegate.get(index);
+            delegate = newDelegate;
+            return previousElement;
+        }
+
+        @SuppressWarnings("unchecked")
+        private boolean setDelegateAndCheckChanged(Seq<?> newDelegate) {
+            final boolean changed = delegate.size() != newDelegate.size();
+            delegate = (Seq<T>) newDelegate;
+            return changed;
+        }
+
+        // -- Javaslang converters
+
+        Vector<T> asVector() {
+            return delegate.toVector();
+        }
+
+        List<T> asList() {
+            return delegate.toList();
+        }
+
+        Array<T> asArray() {
+            return delegate.toArray();
+        }
+
+    }
+
+    @FunctionalInterface
+    private interface ListIteratorFactory<T, S extends Seq<T>> extends BiFunction<S, Integer, ListIterator<T>>, Serializable {
+    }
+
+    static class SetAsJavaSet<T> implements java.util.Set<T>, Serializable {
+        private static final long serialVersionUID = 1L;
+        private Set<T> delegate;
+
+        SetAsJavaSet(Set<T> delegate) {
+            this.delegate = delegate;
+        }
+        @Override
+        public int size() {
+            return TODO();
+        }
+
+        @Override
+        public boolean isEmpty() {
+            return TODO();
+        }
+
+        @Override
+        public boolean contains(Object o) {
+            return TODO();
+        }
+
+        @Override
+        public Iterator<T> iterator() {
+            return TODO();
+        }
+
+        @Override
+        public Object[] toArray() {
+            return TODO();
+        }
+
+        @Override
+        public <T1> T1[] toArray(T1[] a) {
+            return TODO();
+        }
+
+        @Override
+        public boolean add(T t) {
+            return TODO();
+        }
+
+        @Override
+        public boolean remove(Object o) {
+            return TODO();
+        }
+
+        @Override
+        public boolean containsAll(Collection<?> c) {
+            return TODO();
+        }
+
+        @Override
+        public boolean addAll(Collection<? extends T> c) {
+            return TODO();
+        }
+
+        @Override
+        public boolean retainAll(Collection<?> c) {
+            return TODO();
+        }
+
+        @Override
+        public boolean removeAll(Collection<?> c) {
+            return TODO();
+        }
+
+        @Override
+        public void clear() {
+            TODO();
+        }
+
+        // -- Object.*
+
+        @Override
+        public boolean equals(Object o) {
+            return (o == this) || ((o instanceof java.util.Set) && areEqual(delegate, (java.util.Set<?>) o));
```

Comment:
This assumes that the iteration orders of both sets are the same. This should use the same implementation as AbstractSet.equals.